lsp-methods.el

```diff
@@ -329,37 +334,69 @@ interface TextDocumentItem {
 	      :version (lsp--cur-file-version)
 	      :text (buffer-substring-no-properties (point-min) (point-max))))))
 
-(defun lsp--shutdown-cur-workspace ()
-  "Shut down the language server process for ‘lsp--cur-workspace’."
-  (with-demoted-errors "LSP error: %S"
-    (lsp--send-request (lsp--make-request "shutdown" (make-hash-table)) t)
-    (lsp--send-notification (lsp--make-notification "exit" nil)))
-  (lsp--uninitialize-workspace))
-
 ;; Clean up the entire state of lsp mode when Emacs is killed, to get rid of any
 ;; pending language servers.
 (add-hook 'kill-emacs-hook #'lsp--global-teardown)
 
 (defun lsp--global-teardown ()
   (with-demoted-errors "Error in ‘lsp--global-teardown’: %S"
-    (maphash (lambda (_k value) (lsp--teardown-client value)) lsp--workspaces)))
+    (maphash (lambda (_k value) (lsp--teardown-workspace value)) lsp--workspaces)))
 
-(defun lsp--teardown-client (client)
-  (setq lsp--cur-workspace client)
+(defun lsp--teardown-workspace (workspace)
+  (setq lsp--cur-workspace workspace)
   (lsp--shutdown-cur-workspace))
 
+(defun lsp--shutdown-cur-workspace ()
```

Comment:
Nit: I'd prefer not moving this function around, as it only generates a needless diff.